SocialNetwork/SocialNetwork.BLL/Services/Auth/AuthService.cs

```diff
@@ -0,0 +1,124 @@
+using AutoMapper;
+using SocialNetwork.BLL.Contracts;
+using SocialNetwork.BLL.DTO.Users.Request;
+using SocialNetwork.BLL.DTO.Users.Response;
+using SocialNetwork.BLL.Exceptions;
+using SocialNetwork.DAL;
+using SocialNetwork.DAL.Contracts.Users;
+using SocialNetwork.DAL.Entities.Users;
+
+namespace SocialNetwork.BLL.Services.Auth;
+
+public class AuthService : IAuthService
+{
+    private readonly IUserRepository _userRepository;
+
+    private readonly IUserProfileRepository _userProfileRepository;
+
+    private readonly ISaltService _saltService;
+
+    private readonly IPasswordHashService _passwordHashService;
+    private readonly IMapper _mapper;
+
+    public AuthService(IUserRepository userRepository, IUserProfileRepository userProfileRepository,
+        ISaltService saltService, IPasswordHashService passwordHashService, IMapper mapper)
+    {
+        _userRepository = userRepository;
+        _userProfileRepository = userProfileRepository;
+        _saltService = saltService;
+        _passwordHashService = passwordHashService;
+        _mapper = mapper;
+    }
+
+    public async Task<bool> IsLoginAlreadyExists(string login)
+    {
+        var users = await _userRepository.GetAllAsync(u => u.Login == login);
+        return users.Any();
+    }
+
+    public async Task<User> AddUser(User newUser)
+    {
+        var user = await _userRepository.AddAsync(newUser);
+        await _userRepository.SaveAsync();
+        return user;
+    }
+
+    public async Task<User?> GetUserByLogin(string login)
+    {
+        var users = await _userRepository.GetAllAsync(u => u.Login == login);
+        return users.FirstOrDefault();
+    }
+
+    public bool IsLoginValid(string login)
```

Comment:
Use some validation libraries for validation purposes: FluentValidation for example